src/platform/forms-system/src/js/fields/TitleField.jsx

```diff
@@ -1,18 +1,49 @@
 import React from 'react';
+
+import PropTypes from 'prop-types';
 import classNames from 'classnames';
 
-export default function TitleField({ id, title }) {
+export default function TitleField({ id, title, formContext }) {
   const isRoot = id === 'root__title';
-
-  const classes = classNames('schemaform-block-title', {
+  const classes = classNames('schemaform-block-title testing', {
     'schemaform-block-subtitle': !isRoot,
   });
+  const { onReviewPage } = formContext;
+  let reviewTitle;
+
+  const isStringTitle = typeof title === 'string';
+  const isEmptyStringTitle = isStringTitle && title.trim() === '';
 
-  const isEmptyTitle = typeof title === 'string' && title.trim() === '';
+  // Handle title scenarios:
+  // If title is a React element with an <h3>,
+  // it should be made <h4> for the Review page.
+  if (isStringTitle) {
+    reviewTitle = !isEmptyStringTitle ? title : null;
+  } else if (!onReviewPage) {
+    reviewTitle = title;
+  } else if (title.type !== 'h3') {
```

Comment:
hm the other tricky thing about this is:
✅ this works for 'ui:title': <h3>...</h3>,  
X    but not 'ui:title': <> <h3>...</h3></> which is what our titleUI does. 
i'm not sure the best course of action but just thinking about it.

Reply:
oh, good catch, Robert!  i'll try refactoring for recursive search and replace.

Reply:
Refactor done; I'm checking for children recursively now and lowering any header-tag's level by one for the Review page.